pkg/filestore/writer.go

```diff
@@ -0,0 +1,91 @@
+package filestore
+
+import (
+	"crypto"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/cert-manager/csi-lib/metadata"
+	"github.com/cert-manager/csi-lib/storage"
+
+	"github.com/jetstack/cert-manager-csi/pkg/apis/defaults"
+	"github.com/jetstack/cert-manager-csi/pkg/apis/v1alpha1"
+	"github.com/jetstack/cert-manager-csi/pkg/apis/validation"
+)
+
+// Writer wraps the storage backend to allow access for writing data
+type Writer struct {
+	Store storage.Interface
+}
+
+func (w *Writer) WriteKeypair(meta metadata.Metadata, key crypto.PrivateKey, chain []byte, ca []byte) error {
+	attrs, err := defaults.SetDefaultAttributes(meta.VolumeContext)
+	if err != nil {
+		return err
+	}
+	if err := validation.ValidateAttributes(attrs); err != nil {
+		return err
+	}
+
+	keyPEM := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: x509.MarshalPKCS1PrivateKey(key.(*rsa.PrivateKey)),
+		},
+	)
+
+	var nextIssuanceTime time.Time
+	if attrs[v1alpha1.DisableAutoRenewKey] == "true" {
+		// We can assume the pod will not outlive the year 9999
+		nextIssuanceTime = time.Date(9999, time.January, 0, 0, 0, 0, 0, time.UTC)
```

Comment:
question: I agree that this assumption is safe :joy:  However, disabling renewal is different to having a renewal time a long way in the future; disabling renewal allows us to skip the "do we need to renew" check.

Is it not possible to disable renewal entirely?

Reply:
Mr original PR removed the 'disable renewal' functionality altogether, as the csi-lib library does not support this mode of operation. What's the reason for it coming back here @JoshVanL?

Reply:
This is a bit of a wart of the csi-lib itself. The nextIssuance time not being nil is used as a signal inside the csi-lib machinery to know whether the volume is signed. We talked about changing this to have a discrete internal metadata field on the csi-lib side, but I think I would sooner remove this functionality in cert-manager-csi.

https://github.com/cert-manager/csi-lib/blob/fe9b5eff41b7a867e93102b4e7cf7847b40aa23c/manager/manager.go#L489

Reply:
@munnerz Sorry my GitHub is running slow it seems :sweat_smile: 

Happy to remove it ^